app/src/bcsc-theme/features/modal/components/SystemModal.tsx

```diff
@@ -0,0 +1,134 @@
+import { Button, ButtonType, ThemedText, useTheme } from '@bifold/core'
+import { useFocusEffect, useNavigation } from '@react-navigation/native'
+import { useCallback, useRef } from 'react'
+import { useTranslation } from 'react-i18next'
+import { ScrollView, StyleSheet, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import Icon from 'react-native-vector-icons/MaterialIcons'
+
+export interface SystemModalProps {
+  /**
+   * Name of the MaterialIcons icon to display
+   */
+  iconName: string
+  /**
+   * Size of the icon (default: 200)
+   */
+  iconSize?: number
+  /**
+   * Translation key for the header text
+   */
+  headerKey: string
+  /**
+   * Array of translation keys for content paragraphs
+   */
+  contentKeys: string[]
+  /**
+   * Translation key for the button title
+   */
+  buttonTitleKey: string
```

Comment:
nit: probably simpler to pass the string values vs the translation key for these. Would also prevent having to pass the translationParam below.

ps: love that you consolidated these modals into a shared component